classes/Cart.php

```diff
@@ -755,6 +755,19 @@ public function getProducts($refresh = false, $id_product = false, $id_country =
                     $reduction_type_row = ['reduction_type' => 0];
                 }
 
+                /*
+                 * In case of packs, we need to properly calculate the quantity in stock. We can't just use
+                 * the quantity from the query, because stocks can be set to use the quantity of the products
+                 * in them. The quantity in stock_available then has no meaning and could be always zero.
+                 *
+                 * When calling Pack::getQuantity here, you MUST use null for $cart parameter. Otherwise it
+                 * will subtract the quantity that is already in the cart. Basically resulting in a nonsense,
+                 * half of quantity you have. We need the REAL quantity.
+                 */
+                if (Pack::isPack($product['id_product'])) {
+                    $product['quantity_available'] = Pack::getQuantity((int) $product['id_product'], (int) $product['id_product_attribute']);
+                }
+
```

Comment:
This portion of the code is only used in error message in the default theme. And since we are not passing a cart, the result will be the same.

We could theoretically just use `Product::getQuantity((int) $product['id_product'], (int) $product['id_product_attribute'])` here, but why to fetch it one more time, when it would get us the exact same thing as we have. :-)